nats.go

```diff
@@ -624,11 +625,30 @@ type Msg struct {
 	Header  Header
 	Data    []byte
 	Sub     *Subscription
+	// Internal
 	next    *Msg
+	wsz     int
 	barrier *barrierInfo
 	ackd    uint32
 }
 
+// Compares two msgs, ignores sub but checks all other public fields.
+func (m *Msg) Equal(msg *Msg) bool {
+	if m == msg {
+		return true
+	}
+	if m == nil || msg == nil {
+		return false
+	}
+	if m.Subject != msg.Subject || m.Reply != msg.Reply {
+		return false
+	}
+	if !bytes.Equal(m.Data, msg.Data) {
+		return false
+	}
+	return reflect.DeepEqual(m.Header, msg.Header)
```

Comment:
What would you think about implementing `Equal()` method for `Header` instead of using `reflect.DeepEqual()`?? It's quite a lot faster and a bit cleaner IMO:
```go
func (h Header) Equal(h2 Header) bool {
	if len(h) != len(h2) {
		return false
	}
	for k, v1 := range h {
		v2, ok := h2[k]
		if !ok {
			return false
		}
		for i, header := range v1 {
			if header != v2[i] {
				return false
			}
		}
	}
	return true
}
```

Reply:
Yep we can do that.

Reply:
ok done.

Reply:
I was thinking about extracting header comparison to separate `Equal()` method, as a utility, but as long as we do not need it, your approach is also fine.